### DISCUSSION TITLE
Sma sensorbox anemometer graph

### DISCUSSION THREAD

**Discussion by baskarz** [jpcornil-git/HA-sma#8] [Q&A] [answered]
Hi, I have the anemometer connected to my sensorbox, it displays values fine but the history graph is a bar type whereas I would like it to be a line graph like the temperature history graphs.
How would I go about achieving this?
The screenshot attached shows what it currently is, I would like it to be a line graph.
Thanks
![Screenshot_20231028_085915_Chrome](https://github.com/jpcornil-git/HA-sma/assets/52617323/974fab7f-163d-45d9-8622-a22995109ecc)

**Comment by jpcornil-git** [comment, marked as answer]
Hi,

I just checked in changes required to add support for "km/h" speed unit (only "m/s" was supported, unknown units are handled as raw states and therefore displayed as such), see [diff](https://github.com/jpcornil-git/HA-sma/commit/6840891a79c0217d1a4ea52ed7affcb688dcef09).
 
-> could you give it a try and let me know, once tested OK, I'll add a new release tag.

Just checkout top of main in your custom_component/sma_webbox directory or if you are using HACS:
- go to HACS->Integrations->sma_webbox
- click on the 3 dots at the top right and select "Reload"
- enable "select beta versions" and select "main"
- click "Download" and restart HA

You should now see 6840891 iso v2.1.0 as version number (HACS->Integrations->sma_webbox, top left) and hopefully (I can't test myself) the desired graph :-).

**Reply by baskarz** [reply]
Hi,

Awesome, thank you for that!

Worked great.
I can now see the line graph as expected :)
I've attached a screenshot too.

![Screenshot_20231030_093627_Chrome](https://github.com/jpcornil-git/HA-sma/assets/52617323/4e289ea5-3bba-4c8a-b272-46726807f877)

**Reply by baskarz** [reply]
One other thing I've just noticed, so the graph I posted before shows fine but the first graph when you tap on the entity is over a 1 second time period. Can this be adjusted to say 6 or 12 hourly?
The graph I screenshot before is when you go into 'show more'
![Screenshot_20231030_150328_Chrome](https://github.com/jpcornil-git/HA-sma/assets/52617323/37e70ea5-98da-46db-889a-9b87bbf61a1b)

**Comment by jpcornil-git** [comment]
I just released v2.1.1 to add support for km/h speed unit -> you can now update the integration from HACS without checking beta/selecting latest.

Wrt the history graph, this is a priori not related to/controlled by this integration but HA.
Do you have the same when you add an history graph for that entity to one of your dashboard ?
Nothing in your logs that could point to a clue ? 
It could also be related to the data format change (state/no units until yesterday and float/km/h since then) confusing the card -> you may want to delete (or rename) your home-assistant_v2.db to start with no historical data (or wait for 10 days/default retention time for short term data unless you changed that)

**Comment by jpcornil-git** [comment]
Ending the discussion as there is no further feedback
